cmake/onednn_lite_config.cmake

```diff
@@ -0,0 +1,353 @@
+include_guard()
+
+get_property(DNNL_INCLUDES GLOBAL PROPERTY GC_DNNL_INCLUDES)
+get_property(DNNL_PATH GLOBAL PROPERTY GC_DNNL_SOURCE_DIR)
+if (NOT DEFINED DNNL_INCLUDES)
+    return()
+endif ()
+
+########## This cmake build lite version of onednn, containing only microkernel related codes
+
+set(APP_NAME "dnnl_brgemm")
+
+# Build onednn
+set(DNNL_BUILD_TESTS OFF)
+set(DNNL_BUILD_EXAMPLES OFF)
+set(DNNL_ENABLE_JIT_PROFILING OFF)
+set(DNNL_BLAS_VENDOR NONE)
+set(DNNL_LIBRARY_TYPE STATIC)
+
+set(DNNL_GPU_RUNTIME "NONE")
+if(NOT DEFINED DNNL_CPU_RUNTIME)
+    set(DNNL_CPU_RUNTIME "OMP")
+    set(DNNL_CPU_THREADING_RUNTIME "OMP")
+endif()
+
+if(${DNNL_CPU_RUNTIME} STREQUAL "OMP")
+    find_package(OpenMP REQUIRED)
+endif()
+
+if(${DNNL_CPU_RUNTIME} STREQUAL "TBB")
+    include("${DNNL_PATH}/cmake/TBB.cmake")
+endif()
+
+########## copied from main cmake file of DNNL
+# Set the target architecture.
+if(NOT DNNL_TARGET_ARCH)
+    if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64.*|AARCH64.*|arm64.*|ARM64.*)")
+        set(DNNL_TARGET_ARCH "AARCH64")
+    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(ppc64.*|PPC64.*|powerpc64.*)")
+        set(DNNL_TARGET_ARCH "PPC64")
+    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(s390x.*|S390X.*)")
+        set(DNNL_TARGET_ARCH "S390X")
+    else()
+        set(DNNL_TARGET_ARCH "X64")
+    endif()
+endif()
```

Comment:
Here and everywhere else: Is this not handled by the onednn cmake?

Reply:
The reasonings behind this new cmake: 
The newly added BRGEMM runtime used by `microkernel` pipeline depends on dnnl and need to actually build it., however `onednn.cmake` only does some header processing. So I removed those preprocessing cmds and added a new cmake `onednn_lite_config.cmake` to config the build of dnnl (That's quite a long list of configs so I create an isolated file). In addition, this new cmake only build selected files needed by BRGEMM to minimize compile time & binary size.

Reply:
We bring in the sources, why do we need to copy? Can't you just use the related cmake files directly? Include them if necessary?

Reply:
Do you mean we can include cmake in onednn subdirectory? I think there's two problems for that: 1. the finest-grained cmake in onednn is still much larger than what we need; 2. we still need to copy & run the preprocessing in main CMake for that to work;